src/lib/provable/scalar-field.ts

```diff
@@ -0,0 +1,44 @@
+import { createField } from './core/field-constructor.js';
+import { createForeignField } from './foreign-field.js';
+import { field3ToShiftedScalar } from './gadgets/native-curve.js';
+import { Provable } from './provable.js';
+import { Scalar } from './scalar.js';
+
+export { ScalarField };
+
+/**
+ * ForeignField representing the scalar field of Pallas and the base field of Vesta
+ */
+class ScalarField extends createForeignField(
+  28948022309329048855892746252171976963363056481941647379679742748393362948097n
+) {
+  /**
+   * Provable method to conver a {@link ScalarField} into a {@link Scalar}
+   *
+   * This is always possible and unambiguous, since the scalar field is larger than the base field.
+   */
+  public toScalar(): Scalar {
```

Comment:
maybe there should be a static version of this method, which can be used on instances of the base `ForeignField`?
so we don't have to convert with `new ScalarField()` before doing `.toScalar()`?

Reply:
I don't fully understand how putting that function into `ForeignField` would work, since this method can only exist if the modulus is exactly Fq.modulus, which is only true in the `ScalarField`.

Reply:
I mean put a static version of this method on this class